packages/react/src/components/UIShell/SideNavButton.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright IBM Corp. 2025,
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React, {
+  ElementType,
+  ForwardedRef,
+  useContext,
+  Ref,
+  ComponentProps,
+} from 'react';
```

Comment:
- `ElementType` is unused.
- `Ref` is unused.
- `type` imports can be used.

```suggestion
import React, {
  useContext,
  type ComponentProps,
  type ForwardedRef,
} from 'react';
```